### ISSUE TITLE
Align with native Unicode property escapes

### ISSUE THREAD

**Issue by mathiasbynens** [slevithan/xregexp#225]
Other than [the set of supported properties](https://github.com/slevithan/xregexp/pull/220#issuecomment-366521619), there are some key differences between XRegExp’s handling of `\p{…}` and the way they work in native JS.

- Native `\p{…}` doesn’t implement loose matching; only strict, case-sensitive matches for canonical property names and values (or their aliases) are accepted
- In native `\p{…}`, `Block`s are not supported
- Native `\p{…}` doesn’t support for the `In` prefix or any other prefix (although since XRegExp only does this for `Block`s, dropping `Block` support already resolves this)
- Native `\p{…}` supports `Script_Extensions` which is generally more useful than `Script`

Technically these are all breaking changes, but IMHO we should consider aligning with native property escapes.

**Comment by slevithan** [comment]
Supporting Script_Extensions is the biggest change here. It might make sense to just ensure that use of Script_Extensions is passed through correctly and works when in an ES2018 environment that supports it natively. That would avoid the need for adding the large and significantly redundant associated data, but then, removing Blocks would "free up" space for this. I'm definitely open to adding it as a new addon.

The other changes (dropping support for loose name matching and Blocks) should be straightforward. Happy to adopt these changes any time, and will go ahead with them if others don't get to them before me.

**Comment by mathiasbynens** [comment]
> I'm definitely open to adding it as a new addon.

Would it make sense to split up the current properties as separate addons? I.e., have one for `Script`, one for `General_Category`, and one for binary properties? Users can then load only the ones they need, reducing their bundle size and thus minimizing the load +  and parse cost.

**Comment by slevithan** [comment]
I think I'm missing something because that sounds like how it's currently handled. Can you clarify?

**Comment by mathiasbynens** [comment]
Right, not sure what I was thinking with that last comment. I’ll remove it now to avoid spreading misinformation :)

**Comment by slevithan** [comment]
Removed support for Unicode blocks in commit 4860122362c9822f35ab7f2deea7973a5815fcac.
